include/simple-mpc/mpc.hpp

```diff
@@ -171,6 +176,13 @@ namespace simple_mpc
       }
     }
 
+    const Eigen::VectorXd getStateDerivative(const std::size_t t)
+    {
+      ExplicitIntegratorData * int_data =
+        dynamic_cast<ExplicitIntegratorData *>(&*solver_->workspace_.problem_data.stage_data[t]->dynamics_data);
+      return int_data->continuous_data->xdot_;
```

Comment:
Please return `ConstVectorRef` to get a numpy view array in Python (that you can explicitly copy). This avoids doing copies when not needed.

Reply:
Also add an `assert()` or other runtime check that `int_data` is not nullptr - which would be a cast error.